pkg/operator/cleanup/cleanup_controller.go

```diff
@@ -0,0 +1,179 @@
+		log.WithField("cr", credentialRequestName).WithField("namespace", namespace).Info("observed stale credential request event")
+		return true
+	}
+
+	return false
+}
+
+// contains checks if a given credential request is present in a slice of stale credential requests
+func contains(credentialRequests []types.NamespacedName, credentialRequest types.NamespacedName) bool {
+	for _, cr := range credentialRequests {
+		if cr == credentialRequest {
+			return true
+		}
+	}
+	return false
+}
+
+var _ reconcile.Reconciler = &ReconcileStaleCredentialsRequest{}
+
+// ReconcileStaleCredentialsRequest reconciles a stale CredentialsRequest object
+type ReconcileStaleCredentialsRequest struct {
+	ReconcileCredentialsRequest credentialsrequest.ReconcileCredentialsRequest
+}
+
+// Reconcile marks the stale CredentialsRequest object for deletion
+func (r *ReconcileStaleCredentialsRequest) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	start := time.Now()
+
+	logger := log.WithFields(log.Fields{
+		"controller": controllerName,
+		"cr":         fmt.Sprintf("%s/%s", request.NamespacedName.Namespace, request.NamespacedName.Name),
+	})
+
+	defer func() {
+		dur := time.Since(start)
+		metrics.MetricControllerReconcileTime.WithLabelValues(controllerName).Observe(dur.Seconds())
+	}()
+
+	mode, conflict, err := utils.GetOperatorConfiguration(r.ReconcileCredentialsRequest.Client, logger)
+	if err != nil {
+		logger.WithError(err).Error("error checking if operator is disabled")
+		return reconcile.Result{}, err
+	} else if conflict {
+		logger.Error("configuration conflict betwen legacy configmap and operator config")
+		return reconcile.Result{}, fmt.Errorf("configuration conflict")
+	}
+
+	logger.Info("syncing stale credentials request")
+	cr := &minterv1.CredentialsRequest{}
+	err = r.ReconcileCredentialsRequest.Get(context.TODO(), request.NamespacedName, cr)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Debug("credentials request no longer exists")
+			return reconcile.Result{}, nil
+		}
+		logger.WithError(err).Error("error getting credentials request, requeuing")
+		return reconcile.Result{}, err
+	}
+	logger = logger.WithFields(log.Fields{
+		"secret": fmt.Sprintf("%s/%s", cr.Spec.SecretRef.Namespace, cr.Spec.SecretRef.Name),
+	})
+
+	origCR := cr
+	cr = cr.DeepCopy()
+
+	if mode == operatorv1.CloudCredentialsModeManual {
+		logger.Warnf("operator set to disabled / manual mode, user needs to delete stale credentials")
+
+		msg := fmt.Sprintf("CredentialsRequest is no longer required. Delete CR, Secret containing credentials, and associated platform/cloud resources")
+		reason := "CredentialsNoLongerRequired"
+		updateCheck := utils.UpdateConditionIfReasonOrMessageChange
+		status := corev1.ConditionTrue
+
+		cr.Status.Conditions = utils.SetCredentialsRequestCondition(cr.Status.Conditions, minterv1.StaleCredentials,
+			status, reason, msg, updateCheck)
+
+		err := r.ReconcileCredentialsRequest.UpdateStatus(origCR, cr, logger)
```

Comment:
Lets move this UpdateStatus() into a shared location so that ReconcileStaleCredentialsRequest does not need to embed ReconcileCredentialsRequest